src/parser/Expression.ts

```diff
@@ -645,5 +645,112 @@ export class CallfuncExpression implements Expression {
         );
         return result;
     }
+}
+
+export class ConditionalExpression implements Expression {
+    constructor(
+        readonly test: Expression,
+        readonly consequent: Expression,
+        readonly alternate: Expression
+    ) {
+        this.range = Range.create(
+            test.range.start,
+            alternate.range.end
+        );
+    }
 
+    public readonly range: Range;
+
+    transpile(state: TranspileState) {
+        let result = [];
+        let [testExpressions, testVarExpressions, testUniqueNames] = this.getExpressionInfo(this.test);
+        let [consequentExpressions, consequentVarExpressions, consequentUniqueNames] = this.getExpressionInfo(this.consequent);
+        let [alternateExpressions, alternateVarExpressions, alternateUniqueNames] = this.getExpressionInfo(this.alternate);
+
+        let allExpressions = [...testExpressions, ...consequentExpressions, ...alternateExpressions];
+        let allUniqueVarNames = [...new Set([...testUniqueNames, ...consequentUniqueNames, ...alternateUniqueNames])];
+
+        let mutatingExpressions = allExpressions.filter(e => e instanceof CallExpression || e instanceof CallfuncExpression || e instanceof DottedGetExpression);
```

Comment:
We might need to also include object property references as "expensive" enough to run in the deferred function, because rendezvous could be fairly slow, and the "bslib_simpleTernary" would cause TWO rendezvous.

@chrisdp what do you think?

Reply:
any way of knowing that up front? I don't think so, no? coz we'd need to know it's a node type. Does that mean any dotteted, get then, irrespective of the presence function calls?

Reply:
@georgejecook I spoke with @chrisdp and we both agree, there are situations that could cause a rendezvous when accessing object members, so we cannot allow properties to follow the simpleTernary approach since that could be expensive in the long run. So instead we need to restrict the simpleTernary to only variables and literals.

Reply:
No, there's no way of knowing that up front. So yeah, any dotted get would also need to use the scope-capture implementation.